From: Gerald Schaefer <geraldsc@de.ibm.com>
Subject: zfcp: Fix reference counting for cfdc requests
References: bnc#487755,LTC#52849

Symptom:     Setting adapter off-line will hang.
Problem:     The adapter reference count can get negative.
Solution:    Increase the reference count when issuing
             cfdc requests.

Acked-by: John Jolly <jjolly@suse.de>
---
 drivers/s390/scsi/zfcp_cfdc.c |    1 +
 1 file changed, 1 insertion(+)

Index: linux-sles11/drivers/s390/scsi/zfcp_cfdc.c
===================================================================
--- linux-sles11.orig/drivers/s390/scsi/zfcp_cfdc.c
+++ linux-sles11/drivers/s390/scsi/zfcp_cfdc.c
@@ -207,6 +207,7 @@ static long zfcp_cfdc_dev_ioctl(struct f
 		retval = -ENXIO;
 		goto free_buffer;
 	}
+	zfcp_adapter_get(adapter);
 
 	retval = zfcp_cfdc_sg_setup(data->command, fsf_cfdc->sg,
 				    data_user->control_file);
